src/supremm/datasource/prometheus/prominterface.py

```diff
@@ -0,0 +1,476 @@
+import os
+import logging
+import urllib.parse as urlparse
+import datetime
+
+import numpy as np
+import requests
+
+from supremm.config import Config
+
+CHUNK_SIZE = 4 # HOURS
+
+
+class PromClient():
+    """ Client class to interface with Prometheus """
+
+    def __init__(self, resconf):
+        self._url = "http://{}".format(resconf['prom_host'])
+
+        self._client = requests.Session()
+        self._client.mount(self._url, self._client.get_adapter(self._url))
+        self._client.headers.update({'Content-Type': 'application/x-www-form-urlencoded',
+                                     'Accept': 'application/json'})
+
+        self._client.auth = (resconf['prom_user'], resconf['prom_password'])
+        self.connection = PromClient.build_info(self._client, self._url)
+
+    def __str__(self):
+        return self._url
+
+    @staticmethod
+    def build_info(client, test_url):
+        """ Query server build info. Test connection to server. """
+
+        endpoint = "/api/v1/status/buildinfo"
+        url = urlparse.urljoin(test_url, endpoint)
+
+        r = client.get(url)
+        if r.status_code != 200:
+            print(str(r.content))
```

Comment:
Please use logging class rather than `print`

Reply:
Resolved in https://github.com/ubccr/supremm/pull/259/commits/ba29736be72b3035bebd70ae7d0c636ac46ad87b